lib/pengine/bundle.c

```diff
@@ -1585,6 +1663,72 @@ pe__print_bundle(pe_resource_t *rsc, const char *pre_text, long options,
     }
 }
 
+void
+pe__output_bundle(pe_resource_t *rsc, long options, void * out)
+{
+    pcmk__output_t * _out = (pcmk__output_t *)out;
+    pe__bundle_variant_data_t *bundle_data = NULL;
+    char *child_text = NULL;
+    char buffer[LINE_MAX];
+    CRM_CHECK(rsc != NULL, return);
+
+    if (options & pe_print_xml) {
+        bundle_output_xml(rsc, options, out);
+        return;
+    }
```

Comment:
With the output model, there is no longer any need for distinguishing formats. pe__output_bundle() should be identical for all formats (at a high level); anything that needs to be different per format should be in its own custom output function.

Reply:
If we unconditionally call the bundle_output_xml, then we don't need everything that comes after return, so I completely removed the old `pe__output_bundle` and renamed the `bundle_output_xml` into `pe__output_bundle`. The `output_bundle_replica` became unused and I also removed it.

Reply:
This is probably already obvious from later comments, but pe__output_bundle() wouldn't call bundle_output_xml(), it would call out->message(out, "bundle", ...). That would be equivalent to bundle_output_xml() whenever xml output was selected, but later we will add HTML and text implementations as well.